### PR TITLE
Fixed #25421 -- Fixed test --keepdb option on Oracle.

### PR DESCRIPTION
**Pull request by felixxm**
This patch is to avoid two exceptions:
* _ORA-01543 - Tablespace already exists_ (in _execute_test_db_creation),
* _ORA-01920 - User already exists_ (in _create_test_user),

when "--keepdb" is switch on in Oracle tests. In this special case they are acceptable.

### CODE REVIEW

File: django/db/backends/oracle/creation.py [hunk 1]
```diff
@@ -216,7 +227,15 @@ def _create_test_user(self, cursor, parameters, verbosity):
...
-        self._execute_statements(cursor, statements, parameters, verbosity)
+        try:
+            # Statement can fail when keepdb is on
+            self._execute_statements(cursor, statements, parameters, verbosity, allow_quiet_fail=keepdb)
+        except DatabaseError as err:
+            description = str(err)
+            # Avoid user already exists error when keepdb is on
+            if 'ORA-01920' not in description or not keepdb:
+                raise
+
```
Comment (shaib) [inline django/db/backends/oracle/creation.py]: Note that with the two instances you added, there are now 3 instances in the file of almost the same pattern: 
```python
        try:
            # Statement can fail when keepdb is on
            self._execute_statements(cursor, statements, parameters, verbosity, allow_quiet_fail=keepdb)
        except DatabaseError as err:
            description = str(err)
            # Avoid user already exists error when keepdb is on
            if specific_error not in description or not keepdb:
                raise
```
I think the 3 instances can be folded into `_execute_statements()` (and your new instances can also print a message when execution fails acceptably, when `verbosity>=2`, like the existing one does).

**Comment by shaib** [comment]
Other than that, LGTM.

File: django/db/backends/oracle/creation.py [hunk 2]
```diff
@@ -75,8 +75,12 @@ def _create_test_db(self, verbosity=1, autoclobber=False, keepdb=False):
             if verbosity >= 1:
                 print("Creating test user...")
             try:
-                self._create_test_user(cursor, parameters, verbosity)
+                self._create_test_user(cursor, parameters, verbosity, keepdb)
             except Exception as e:
+                # if we want to keep the db then we want also to keep user,
```
Comment (timgraham) [inline django/db/backends/oracle/creation.py]: I see you copy/pasted but I think "If we want to keep the db, then we also want to keep the user." is sufficient for this comment.

**Comment by timgraham** [comment]
Please squash your commits when updating and use the commit message from the pull request title (adapted to fit the guidelines in our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist)).

**Comment by timgraham** [comment]
I think we could backport this to 1.8 too, if you could add a note in the 1.8.5 release notes.

**Comment by shaib** [comment]
Replaced by #5362

**Pull request closed without merging**